contracts/dtoken/src/user_flow_protection.rs

```diff
@@ -0,0 +1,125 @@
+use near_sdk::borsh::{self, BorshDeserialize, BorshSerialize};
+use near_sdk::{AccountId, BlockHeight, log, near_bindgen};
+use near_sdk::collections::{UnorderedMap};
+use near_sdk::env::block_height;
+
+const BLOCKS_TO_NEXT_OPERATION: BlockHeight = 100;
+
+#[near_bindgen]
+#[derive(BorshDeserialize, BorshSerialize)]
+pub struct ActionMutex {
+    blocked_accounts: UnorderedMap<AccountId, BlockHeight>,
```

Comment:
Do we really need to use UnorderedMap ? It's heavy and if you don't need to use iterator it's better to use LookupMap